### ISSUE TITLE
Epic Games Store offline

### ISSUE THREAD

**Issue by MonsterBackup** [FriendsOfGalaxy/galaxy-integration-epic#5]
Galaxy says that Epic Games is offline even when I have it open and online. And every time I click retry, it happens again.

**Comment by Zusier** [comment]
Same issue, 2FA enabled, disabled, Xbox login, regular login, google login. All offline.

**Comment by valsu** [comment]
Same issue here :-(
![2020-05-31_08h37_35](https://user-images.githubusercontent.com/12234737/83346110-fcdbe200-a319-11ea-9903-09e0be3c5d3e.png)

**Comment by eugenkoenig** [comment]
Same for me.

**Comment by Link28x** [comment]
Same here, tried re installing, using my wife's account and mine, error persists

**Comment by nightflyerkilo** [comment]
Same issue. I just installed Galaxy for the first time and encountered this immediately after authenticating with Epic.

**Comment by Varstahl** [comment]
Same, today I've happened to update a bunch of integrations, and epic makes me login everytime, it works all the time, but immediately is reported as offline.
```
galaxy.api.jsonrpc - INFO - Handling request: id=17, method=pass_login_credentials, params={'cookies': '****', 'credentials': '****', 'step': ''}
galaxy.task_manager - DEBUG - Task manager jsonrpc server: creating task 1 (pass_login_credentials)
galaxy.task_manager - DEBUG - Task manager plugin external: creating task 1 (pass_login_credentials)
galaxy.http - WARNING - Got status 404 while performing GET request for https://www.epicgames.com/id/api/exchange
root - DEBUG - Sending data: {"jsonrpc": "2.0", "id": "17", "error": {"code": 0, "message": "Unknown error"}}
galaxy.task_manager - DEBUG - Task manager jsonrpc server: finished task 1 (pass_login_credentials)
galaxy.api.jsonrpc - INFO - Handling request: id=21, method=init_authentication, params={}
galaxy.task_manager - DEBUG - Task manager jsonrpc server: creating task 2 (init_authentication)
galaxy.task_manager - DEBUG - Task manager plugin external: creating task 2 (init_authentication)
galaxy.task_manager - DEBUG - Task manager plugin external: finished task 2 (init_authentication)
root - DEBUG - Sending data: {"jsonrpc": "2.0", "id": "21", "result": {"next_step": "web_session", "auth_params": {"window_title": "Login to Epic\u2122", "window_width": 580, "window_height": 750, "start_uri": "https://www.epicgames.com/id/login", "end_uri_regex": ".*https://www\\.epicgames\\.com/id/api/redirect\\?.*"}}}
galaxy.task_manager - DEBUG - Task manager jsonrpc server: finished task 2 (init_authentication)
galaxy.api.jsonrpc - INFO - Handling request: id=25, method=pass_login_credentials, params={'cookies': '****', 'credentials': '****', 'step': ''}
galaxy.task_manager - DEBUG - Task manager jsonrpc server: creating task 3 (pass_login_credentials)
galaxy.task_manager - DEBUG - Task manager plugin external: creating task 3 (pass_login_credentials)
galaxy.http - WARNING - Got status 404 while performing GET request for https://www.epicgames.com/id/api/exchange
root - DEBUG - Sending data: {"jsonrpc": "2.0", "id": "25", "error": {"code": 0, "message": "Unknown error"}}
galaxy.task_manager - DEBUG - Task manager jsonrpc server: finished task 3 (pass_login_credentials)
```
[plugin-epic-logs.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-epic/files/4708652/plugin-epic-logs.zip)

**Comment by AyoCJ** [comment]
Just did a fresh install of Windows 10 today and now I'm having this same issue, was fine last night

**Comment by valsu** [comment]
> 
> 
> Just did a fresh install of Windows 10 today and now I'm having this same issue, was fine last night

That gives me an idea... Maybe the issue is related to the recent Windows 10 update to v2004?

**Comment by AyoCJ** [comment]
> > Just did a fresh install of Windows 10 today and now I'm having this same issue, was fine last night
> 
> That gives me an idea... Maybe the issue is related to the recent Windows 10 update to v2004?

Could be, I was on 1909 now on 2004

**Comment by Zusier** [comment]
I was on 1809 and now 1909, doesnt work.

**Comment by AyoCJ** [comment]
Reinstalled GOG to try and fix another issue and tried integrating EGS first and it worked, I probably should've tried that earlier

**Comment by Varstahl** [comment]
I've been on 2004 since the insider preview, so I doubt that's what's causing the issue. Also it seems an issue with the login response rather than the rest of the plugin itself. And, out of pure speculation, I just reinstalled GoG and the EGS plugin, and it doesn't work, so that's not an option.

**Edit:** I just read back my log, and well, `404` on `https://www.epicgames.com/id/api/exchange` ain't no way this is going to work anymore. Gotta check the new endpoint I guess.

**Comment by Varstahl** [comment]
Yep, `/id/api/exchange/generate` that I just found works out, also 0.48.2 that's just been released fixed the issue. Thanks, @FriendsOfGalaxy, seems I was a couple hours too late :)

**Comment by Zusier** [comment]
Yep, latest release works, thanks @FriendsOfGalaxy!

**Comment by valsu** [comment]
> 
> 
> Yep, `/id/api/exchange/generate` that I just found works out, also 0.48.2 that's just been released fixed the issue. Thanks, @FriendsOfGalaxy, seems I was a couple hours too late :)
It's fixed!

**Comment by Link28x** [comment]
Confirmed Thanks!!!

**Comment by AmitKhare1088** [comment]
Hello @FriendsOfGalaxy,

After going through the comments, it looks like this issue is solved for some users, but this issue started for me after the latest update. Hard to say whether it was Windows 10 2004 update or Epic integration update.
The plugin keeps crashing and disconnecting. I even reinstalled gog but it didn't work :(
I have attached the logs. Please take a look.

Thank you for your work.

[plugin-epic-24d5f405-7211-4198-83c4-50a6f63b1dcf.log](https://github.com/FriendsOfGalaxy/galaxy-integration-epic/files/4714589/plugin-epic-24d5f405-7211-4198-83c4-50a6f63b1dcf.log)

**Comment by FriendsOfGalaxy** [comment]
@AmitKhare1088  At first glance it doesnt seem that anything is wrong with these logs, i will investigate further and let you know.

**Comment by AmitKhare1088** [comment]
@FriendsOfGalaxy  Thank you :)

**Comment by Balto-o** [comment]
Hi,
Since the latest update (v0.48.2) I've also been encountering constant plugin crashing.
I don't know where to find logs tho.

**Comment by AmitKhare1088** [comment]
@Balto92  You can go to - C:\ProgramData\GOG.com\Galaxy\logs
 and find the log for epic, it should start with plugin-epic.

**Comment by Balto-o** [comment]
Thank you.
Here's what I've found.
[plugin-epic-24d5f405-7211-4198-83c4-50a6f63b1dcf.log](https://github.com/FriendsOfGalaxy/galaxy-integration-epic/files/4724201/plugin-epic-24d5f405-7211-4198-83c4-50a6f63b1dcf.log)

**Comment by AmitKhare1088** [comment]
@FriendsOfGalaxy  Although I am having an issue as others stated, it does not occur if EGS starts up when PC is booted and keeps running in the background. The plugin crashes after few minutes if its not running in the background. Can someone else confirm? @Balto92

**Comment by Balto-o** [comment]
In my case the plugin crashes even with EGS running at startup

**Comment by Varstahl** [comment]
@AmitKhare1088 not on my end, no. Loading/Exiting EGS has no effect on the loaded plugin, at least with plugin version 0.48.2 and latest (april's?) version of Galaxy 2.
Also @Balto92 your error seems pretty weird and might be related to your own environment rather than the plugin itself. That's a pretty strange error on a library.

**Comment by AmitKhare1088** [comment]
@Balto92 @Varstahl  Thanks for confirming! hope this gets resolved soon.

**Comment by FriendsOfGalaxy** [comment]
@AmitKhare1088 @Balto92 
you've probably encounter different issue, that should be already fixed in : #6 
Hope that helps!
